quickwit/quickwit-serve/src/search_api/rest_handler.rs

```diff
@@ -58,9 +58,10 @@ pub struct SearchApi;
 pub(crate) async fn extract_index_id_patterns(
     comma_separated_index_patterns: String,
 ) -> Result<Vec<String>, Rejection> {
-    let index_pattern = percent_decode_str(&*comma_separated_index_patterns)
+    let index_pattern = percent_decode_str(&comma_separated_index_patterns)
         .decode_utf8()
-        .unwrap();
+        .expect("the index_pattern should be url decoded")
```

Comment:
can you use `.context("index pattern does not contain valid utf8 characters")` instead of `expect`?